my_project/my_app/views.py

```diff
@@ -1,23 +1,92 @@
     """
-    Sample view class.
+    This is an example of adding/modifying slots using business logic, using
+    the `hotel_booking` competency. There is also an example
+    for arbitrary state transitions.
     """
     def post(self, request, _format=None):  # pylint: disable=unused-argument, no-self-use
         """
         Sample post request function.
-
-        Accepts a post request with two integers
-        and returns a reponse containing their sum.
         """
-        int1, int2 = (int(value) for value in request.data.values())
-        result = int1 + int2
-        return Response(result)
+        # Assign hotel_booking variables
+        if '_TRANSACTION_TYPE_' in request.data['slots']:
+            transaction_type = request.data['slots']['_TRANSACTION_TYPE_']['candidates'][0]['tokens']
+        else:
+            transaction_type = None
+
+        if '_LOCATION_' in request.data['slots']:
+            location = request.data['slots']['_LOCATION_']['candidates'][0]['tokens']
+        else:
+            location = None
+
+        if '_PRICE_' in request.data['slots']:
+            price = request.data['slots']['_PRICE_']['candidates'][0]['tokens']
+        else:
+            price = None
+
+        # this loop sets all of the _SLOTS_ to have a `"resovled": 1` so they will be kept
+        # through each turn of the conversation.  Currently, each turn the slots are sent
+        # with a `"resolved": -1`, so they need to be reset each time, however, they are
+        # changing to be persistent based on their resolved status in an update coming soon
+        for (slot, slot_data) in request.data['slots'].iteritems():
+            if 'candidates' in request.data['slots'][slot]:
+                for candidate in range(len(slot_data['candidates'])):
+                    request.data['slots'][slot]['candidates'][candidate]['resolved'] = 1
+                    if slot <> '_DATE_':
+                        request.data['slots'][slot]['candidates'][candidate]['value'] = \
+                            request.data['slots'][slot]['candidates'][candidate]['tokens']
+            else:
+                request.data['slots'][slot]['resolved'] = 1
+
+        #magical API call to check their credit
+        available_credit = check_available_credit()
+
+        # state transition example
+        # if someone does not have enough credit available to pay for the hotel, 
+        # redirect them to a credit_card_offer state, and return the payload
+        if available_credit < price:
+            request.data['state'] = 'credit_card_offer'
+            return Response(request.data)
+
+        if transaction_type == 'express deal':
+            if location and price:
+                # This is our magical API call to find express deals
+                hotel = find_express_deal(location, price)
+                if hotel:
+                    # This is how to add new _SLOTS_ to the business logic json
+                    _HOTEL_RATING_ = {
```

Comment:
I know you are trying to mimic the slot naming convention but this is a pretty weird variable name. I'd suggest either `HOTEL_RATING` or `hotel_rating`.